src/IB/IBRedundantInitializer.cpp

```diff
@@ -147,6 +147,43 @@ IBRedundantInitializer::getLevelHasLagrangianData(const int level_number, const
     return !d_num_vertex[level_number].empty();
 } // getLevelHasLagrangianData
 
+bool
+IBRedundantInitializer::getIsAllLagrangianDataInDomain(const Pointer<PatchHierarchy<NDIM> > hierarchy) const
+{
+    Pointer<CartesianGridGeometry<NDIM> > grid_geom = hierarchy->getGridGeometry();
+    const double* const domain_x_lower = grid_geom->getXLower();
+    const double* const domain_x_upper = grid_geom->getXUpper();
+
+    for (unsigned int vertex_level_number = 0; vertex_level_number < d_num_vertex.size(); ++vertex_level_number)
+    {
+        Pointer<PatchLevel<NDIM> > level = hierarchy->getPatchLevel(vertex_level_number);
+        const IntVector<NDIM>& ratio = level->getRatio();
+        const IntVector<NDIM>& periodic_shift = grid_geom->getPeriodicShift(ratio);
+
+        for (int j = 0; j < int(d_num_vertex[vertex_level_number].size()); ++j)
+        {
+            for (int k = 0; k < d_num_vertex[vertex_level_number][j]; ++k)
+            {
+                // Points that are inside the domain with periodicity are OK -
+                // exclude points that are truly outside.
+                std::pair<int, int> point_index(j, k);
+                const Point& X = getShiftedVertexPosn(
+                    point_index, vertex_level_number, domain_x_lower, domain_x_upper, periodic_shift);
+
+                for (int d = 0; d < NDIM; ++d)
+                {
+                    if ((X[d] < domain_x_lower[d]) || (domain_x_upper[d] < X[d]))
```

Comment:
Are there any tests that intentionally put IB points right on the domain boundary? It would be good to confirm that it is OK to do this.

At least away from computational domain boundaries, IB points are assigned to patches if patch_lower[d] <= X[d] < patch_upper[d]. We should confirm that this is handled correctly for patches that abut the domain boundary.